runtime-version/lib/MaxAge.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* A max age wrapper. */
+class MaxAge {
+
+  /**
+   * Returns new MaxAge with value 0.
+   *
+   * @returns {MaxAge}
+   */
+  static zero() {
+    return new MaxAge(0, 0);
+  }
+
+  /**
+   * Parses a max-age header string and returns a new MaxAge instance.
+   *
+   * @param {string} str a max-age header string
+   * @returns {MaxAge}
+   */
+  static parse(string) {
+    const match = string.match(/max-age=([0-9]+)[^0-9]?/i);
+    let maxAge = 0;
+    if (match) {
+      maxAge = parseInt(match[1], 10);
+    }
+    return new MaxAge(Date.now(), maxAge);
```

Comment:
What's effectively the difference between this and MaxAge.zero()? Could it be worth to create a MaxAge.now(), and re-use it when checking for the null string check above?

Reply:
made this more explicit